src/udiskslinuxblock.c

```diff
@@ -2656,6 +2656,27 @@ add_blocksize (gchar        **command,
   return TRUE;
 }
 
+static gchar *
+build_command (const gchar *template,
+               const gchar *device,
+               const gchar *label,
+               const gchar *options,
+               GError     **error)
+{
+  gchar *tmp, *tmp2, *command;
+  tmp = udisks_daemon_util_subst_str_and_escape (template, "$DEVICE", device);
+  tmp2 = udisks_daemon_util_subst_str_and_escape (tmp, "$LABEL", label != NULL ? label : "");
+  command = udisks_daemon_util_subst_str (tmp2, "$OPTIONS", options != NULL ? options : "");
```

Comment:
This is ugly, but I don't see a better way to do it. Maybe there's some lightweight templating library we should switch to? Anyway, that's not the target here.

Reply:
asprintf?

Reply:
> asprintf?

That's not really the same thing as replacing values in templates, but it looks to me like the ``$something`` syntax here is an overkill anyway. Thanks for the tip!